feedback-module/src/services/requestService.js

```diff
@@ -1,10 +1,17 @@
 import axios from "axios";
 
-const requestService = (apiEndpoint, obj) => {
-  axios
-    .post(`${process.env.REACT_APP_BACKEND_ENDPOINT}/api/${apiEndpoint}`, obj)
-    .then((res) => console.log(res))
-    .catch((err) => console.log(err));
-};
+async function requestService(apiEndpoint, obj) {
```

Comment:
hmmm the logic for using `requestService` in `Module.js` seems a bit over-complicated— instead of adding the `.then` and `.catch` here, could we just do

```
return await axios.post(`.${process.env.REACT_APP_BACKEND_ENDPOINT}/api/${apiEndpoint}`, obj)```
```

Then in lines 130-144 of `Module`, we can just do something like this
```
await sendFormData(screen.formID).then((res) => {
        console.log(res);
       setScreen(SCREENS[nextScreen]),
            setCheckboxError(false),
            setFailedRequest(false);
      }).catch(err => {setFailedRequest(true);}).finally(() => setLoading(false));
```

Reply:
Just tried this but it doesn't work. Why is it better to throw an error?

Reply:
We could even simplify it even further by giving requestService success and failure callback functions as their arguments:
```
function requestService(apiEndpoint, obj, success, failure, final) {
  axios.post(
    `${process.env.REACT_APP_BACKEND_ENDPOINT}/api/${apiEndpoint}`,
    obj
  ).then(success).catch(failure).finally(final);
}
```

We can then give `sendFormData` these arguments as well and pass them in when we call requestService.

And then instead of making `submitForm` an async function, we could just call `sendFormData` kind of like this:
```
sendFormData(screen.formID), (res) => {
          console.log(res);
          setScreen(SCREENS[nextScreen]),
            setCheckboxError(false),
            setFailedRequest(false);
          setLoading(false);
        }, (err) => {
          console.log(err);
          setFailedRequest(true);
          setLoading(false);
        });
```

Reply:
Then neither of these are aync functions?

Reply:
Also where do success, failure, and final come from?

Reply:
Yes, and these functions could be passed in as arguments to sendFormData (see the last code block).

I think it kind of depends on if we want to have the .then and .catch in requestService though? Not sure what the best practice is for this.

Reply:
> Just tried this but it doesn't work. Why is it better to throw an error?

Sorry didn't see this before— I feel like it's just a little round-a-bout to return a string?